src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -35,19 +45,116 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
+            throw new ParseException(Level.MESSAGE_CONSTRAINTS);
+        }
+        return Level.createLevel(trimmedLevel);
+    }
+
+    public static NextOfKin parseNextOfKin(String nextOfKin) throws ParseException {
+        requireNonNull(nextOfKin);
+        String trimmedNextOfKin = nextOfKin.trim();
+        if (!NextOfKin.isValidNextOfKin(trimmedNextOfKin)) {
+            throw new ParseException(NextOfKin.MESSAGE_CONSTRAINTS);
+        }
+        return new NextOfKin(trimmedNextOfKin);
+    }
+
+    public static Qualification parseQualification(String qualification) throws ParseException {
+        requireNonNull(qualification);
+        String trimmedQualification = qualification.trim();
+        if (!Qualification.isValidQualification(trimmedQualification)) {
+            throw new ParseException(Qualification.MESSAGE_CONSTRAINTS);
+        }
+        return new Qualification(trimmedQualification);
+    }
+
+    public static Institution parseInstitution(String institution) throws ParseException {
+        requireNonNull(institution);
+        String trimmedInstitution = institution.trim();
+        if (!Institution.isValidInstitution(trimmedInstitution)) {
+            throw new ParseException(Institution.MESSAGE_CONSTRAINTS);
+        }
+        return new Institution(trimmedInstitution);
+    }
+
+    public static Subject parseSubject(String subject) throws ParseException {
+        requireNonNull(subject);
+        String trimmedSubject = subject.trim();
+        if (!Subject.isValidSubject(trimmedSubject)) {
+            throw new ParseException(Subject.MESSAGE_CONSTRAINTS);
+        }
+        return Subject.createSubject(trimmedSubject);
+    }
+
+    public static Day parseDay(String day) throws ParseException {
+        requireNonNull(day);
+        String trimmedDay = day.trim();
+        if (!Day.isValidDay(trimmedDay)) {
+            throw new ParseException(Day.MESSAGE_CONSTRAINTS);
+        }
+        return Day.createDay(trimmedDay);
+    }
+
+    public static Time parseTime(String time) throws ParseException {
+        requireNonNull(time);
+        String trimmedTime = time.trim();
+        if (!Time.isValidTime(trimmedTime)) {
+            throw new ParseException(Time.MESSAGE_CONSTRAINTS);
+        }
+        Time resultingTime;
+        try {
+            String[] splitTime = trimmedTime.split("-");
+            resultingTime = new Time(splitTime[0], splitTime[1]);
+        } catch (DateTimeParseException e) {
+            throw new ParseException(Time.MESSAGE_CONSTRAINTS);
+        }
+        return resultingTime;
+    }
+
     /**
      * Parses a {@code String name} into a {@code Name}.
      * Leading and trailing whitespaces will be trimmed.
      *
      * @throws ParseException if the given {@code name} is invalid.
      */
-    public static Name parseName(String name) throws ParseException {
+    public static seedu.address.model.person.Name parsePersonName(String name) throws ParseException {
```

Comment:
Maybe import person.Name and tuitionclass.Name?

Reply:
same here uh got 2 Name in scope conflicting namespace

Reply:
ok!